integration-tests/src/integrationTest/org/apache/cassandra/sidecar/health/SidecarPeerDownDetectorIntegrationTest.java

```diff
@@ -0,0 +1,201 @@
+import org.apache.cassandra.testing.ClusterBuilderConfiguration;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.assertj.core.api.Assertions.assertThat;
+
+class SidecarPeerDownDetectorIntegrationTest extends SharedClusterSidecarIntegrationTestBase
+{
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SidecarPeerDownDetectorIntegrationTest.class);
+
+    List<TestSidecarHostInfo> sidecarServerList = new ArrayList<>();
+
+    @Override
+    protected ClusterBuilderConfiguration testClusterConfiguration()
+    {
+        return new ClusterBuilderConfiguration().nodesPerDc(3);
+    }
+
+    @Override
+    protected void startSidecar(ICluster<? extends IInstance> cluster) throws InterruptedException
+    {
+        Supplier<List<TestSidecarHostInfo>> supplier = () -> sidecarServerList;
+        PeersModule peersModule = new PeersModule(supplier);
+        for (IInstance cassandraInstance : cluster)
+        {
+            // Provider de una lista de Sidecar servers
+            LOGGER.info("Starting Sidecar instance for Cassandra instance {}",
+                        cassandraInstance.config().num());
+            Server server = startSidecarWithInstances(List.of(cassandraInstance), peersModule);
+            sidecarServerList.add(new TestSidecarHostInfo(cassandraInstance, server, server.actualPort()));
+        }
+
+        assertThat(sidecarServerList.size()).as("Each Cassandra Instance will be managed by a single Sidecar instance")
+                                            .isEqualTo(cluster.size());
+
+
+        // assign the server to the first instance
+        server = sidecarServerList.get(0).getServer();
+    }
+
+    class PeersModule extends AbstractModule
+    {
+        Supplier<List<TestSidecarHostInfo>> supplier;
+        public PeersModule(Supplier<List<TestSidecarHostInfo>> supplier)
+        {
+            this.supplier = supplier;
+        }
+
+        @Provides
+        @Singleton
+        @Named("sidecarInstanceSupplier")
+        public Supplier<List<TestSidecarHostInfo>> supplier()
+        {
+            return supplier;
+        }
+
+        @Provides
+        @Singleton
+        public SidecarPeerHealthConfiguration sidecarPeerHealthConfiguration()
+        {
+            return new SidecarPeerHealthConfigurationImpl(true,
+                                                          new MillisecondBoundConfiguration(1, TimeUnit.SECONDS),
+                                                          1,
+                                                          new MillisecondBoundConfiguration(500, TimeUnit.MILLISECONDS));
+        }
+    }
+
+    void stopSidecarInstanceForTest(int instanceId) throws Exception
+    {
+        assertThat(sidecarServerList).isNotEmpty();
+        Server server = sidecarServerList.get(instanceId).getServer();
+        server.stop(serverDeploymentIds.get(server)).toCompletionStage().toCompletableFuture().get(30, TimeUnit.SECONDS);
```

Comment:
nit: you can use the `getBlocking` utility, which block the future and provides richer context on failure.
```suggestion
        String deploymentId = serverDeploymentIds.get(server);
        getBlocking(server.stop(serverDeploymentIds.get(server)), 30, TimeUnit.SECONDS, "Stopping server " + deploymentId);
```